### PR TITLE
[BugFix] Fix iceberg table cannot create expression literal when column type is boolean

### LINKED ISSUE #15733: Iceberg table cannot create expression literal from java.lang.Byte: 1
### Steps to reproduce the behavior (Required)
hive 
1. create table hive_prod.sr_orc_02(
C_BOOL boolean,
c_INT int,
C_date date,
c_String string
)
USING iceberg
TBLPROPERTIES('write.format.default'='orc','write.parquet.compression-codec'='snappy');

2. hive insert
 insert into hive_prod.iceberg_db.sr_13908_orc select true,1,cast('2022-08-08' as date),'Hello, China!';
3.starrocks create catalog
create external catalog ice_catalog01  properties("type"="iceberg","iceberg.catalog.type"="hive","iceberg.catalog.hive.metastore.uris"="thrift://hadoop3:9083");
4.starrocks select 
select * from ice_catalog01.hive_prod.sr_orc_02 where c_bool in (1);
select * from ice_catalog01.hive_prod.sr_orc_02 where c_bool=1;

### Expected behavior (Required)
return correct result 
### Real behavior (Required)
image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by leoyy0316**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15733

## Problem Summary(Required) ：

Solve iceberg talbe binarypredicate issue when column type is boolean

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java [hunk 1]
```diff
@@ -156,7 +160,16 @@ public Expression visitBinaryPredicate(BinaryPredicateOperator operator, Iceberg
             if (columnName == null) {
                 return null;
             }
+
             Object literalValue = getLiteralValue(operator.getChild(1));
+            try {
+                if (context.getSchema().fieldType(columnName).typeId() == Type.TypeID.BOOLEAN) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:166]: move it to `getLiteralValue()`

Comment (leoyy0316) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java]: done

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java [hunk 2]
```diff
@@ -228,6 +237,17 @@ private static Object getLiteralValue(ScalarOperator operator) {
         return operator.accept(new ExtractLiteralValue(), null);
     }
 
+    private static Object convertBoolLiteralValue(Object literalValue, Type.TypeID type) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:240]: put if to outside

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 11 (72.73%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java|8|11|72.73%|[200, 247, 248]|

**Review by stephen-shelby — APPROVED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**